src/cc/bcc_proc.c

```diff
@@ -82,6 +83,40 @@ int bcc_mapping_is_file_backed(const char *mapname) {
     STARTS_WITH(mapname, "[vsyscall]"));
 }
 
+/*
+Finds a file descriptor for a given inode if it's a memory-backed fd.
+*/
+static char *_procutils_memfd_path(const int pid, const uint64_t inum) {
+  char path_buffer[PATH_MAX + 1];
+  char *path = NULL;
+  char *dirstr;
+  DIR *dirstream;
+  struct stat sb;
+  struct dirent *dent;
+
+  snprintf(path_buffer, (PATH_MAX + 1), "/proc/%d/fd", pid);
+  dirstr = malloc(sizeof(char) * (strlen(path_buffer) + 1));
+  strcpy(dirstr, path_buffer);
+  dirstream = opendir(dirstr);
```

Comment:
maybe you want to test dirstream is NULL or not here? If it is NULL, you can just return NULL. This seems more cleaner then below trying to test `dirstream != NULL` in every while loop iteration.

Reply:
I agree